aidants_connect_web/views/new_mandat.py

```diff
@@ -119,31 +119,44 @@ def new_mandat_recap(request):
             }
             mandat_duree = days_before_expiration_date.get(connection.duree_keyword)
             try:
-                # Add a Journal 'create_mandat_print' action
+                # Add a Journal 'create_attestation' action
                 connection.demarches.sort()
-                Journal.objects.mandat_print(
+                Journal.objects.attestation(
                     aidant=aidant,
                     usager=usager,
                     demarches=connection.demarches,
                     duree=mandat_duree,
                     access_token=connection.access_token,
-                    mandat_print_hash=generate_mandat_print_hash(
+                    attestation_hash=generate_attestation_hash(
                         aidant, usager, connection.demarches, mandat_expiration_date
                     ),
                 )
 
-                # The loop below creates one Mandat object per Démarche in the form
+                # Create a mandat
+                mandat = Mandat.objects.create(
+                    organisation=aidant.organisation,
+                    usager=usager,
+                    expiration_date=mandat_expiration_date,
+                    is_remote=True,
+                    duree_keyword=connection.duree_keyword,
+                )
+                # This loop creates one `autorisation` object per `démarche` in the form
                 for demarche in connection.demarches:
-                    Mandat.objects.update_or_create(
+                    # Revoke existing demarche autorisations
+                    Autorisation.objects.filter(
+                        aidant__organisation=aidant.organisation,
+                        usager=usager,
```

Comment:
Ces éléments ne sont plus dans l'autorisation mais dans le mandat : 
```suggestion
                        mandat__organisation=aidant.organisation,
                        mandat__usager=usager,
```

Reply:
Il faut aussi vérifier que les autorisations précédentes ne sont ni expirées, ni révoquées.